bob-backend/src/pearl/disk_controller.rs

```diff
@@ -561,30 +565,26 @@ impl DiskController {
         force_delete: bool,
     ) -> Result<u64, Error> {
         if *self.state.read().await == GroupsState::Ready {
-            let vdisk_group =
-                if !force_delete {
-                    match self.find_group(&op).await {
-                        Ok(group) => group,
-                        Err(_) => return Ok(0)
-                    }
-                } else {
-                    // we should create group only when force_delete == true
+            if let Some(result) = self.get_alien_results(&op, 
+                |g| async move { g.delete(key, meta, StartTimestampConfig::new(false), force_delete).await },
+                0, |s, n| s + n).await {
```

Comment:
Error handling logic should be different here. `delete_alien` should return error if any deletion ended with error (it should process all groups, store the latest error and return it)